packages/shared/src/components/withFeaturesBoundary.tsx

```diff
@@ -1,12 +1,20 @@
 import React, { ComponentType, ReactElement, useContext } from 'react';
 import { FeaturesReadyContext } from './GrowthBookProvider';
+import { useHostStatus } from '../hooks/useHostPermissionStatus';
+import { checkIsExtension } from '../lib/func';
 
 const withFeaturesBoundary = <Props,>(
   WrappedComponent: ComponentType<Props>,
   options?: { fallback?: ReactElement },
 ): ComponentType<Props> => {
   const WithFeaturesBoundary = (props: Props): ReactElement => {
     const { ready: areFeaturesReady } = useContext(FeaturesReadyContext);
+    const { hostGranted } = useHostStatus();
+    const isExtension = checkIsExtension();
+
+    if (isExtension && !hostGranted) {
+      return <WrappedComponent {...props} />;
+    }
```

Comment:
Is this needed?
Just not quite sure why we need it in this spot already

Reply:
Actually, I think there is a better way to do this by just using the fallback and not worrying about extension or host permissions inside this wrapper